File: 2016/Basico-Iteracao-Condicional/fatorial_nao_recursivo.c
#include <stdio.h>
#include <stdlib.h>
/*  Função : Funcao fatorial nao-recursiva
    Autor : Edkallenn
    Data : 06/04/2012
    Observações:
*/

main(){
    int cont,num,k;
    double fat=1;

    printf("Digite um numero inteiro nao negativo: ");
    scanf("%d", &num);
    for (k=1;k<=num;k++){   //loop de 1 ate num
        //calculo do fatorial
        for(cont=1;cont<=k;cont++){
            fat=fat*cont;
        }
        //exibe o fatorial
        printf("O fatorial de %d eh %f\n", k, fat);
        //reinicia a variavel acumuladora
        fat=1;
    }
    getch();
}

File: UNP/ref/Python/Funcoes/FatorialNaoRecursivo.py
def fatorial(n):
    n = n if n > 1 else 1
    j = 1
    for i in range(1, n + 1):
        j = j * i
    return j

def pedir_numero():
    numero = int(input('Digite um numero inteiro nao negativo:'))
    return numero

def tela_boas_vindas():
    print("*********************************")
    print("***   Bem vindo ao programa!  ***")
    print("*********************************")

# Testando...
tela_boas_vindas()
num = pedir_numero()
for i in range(1, num+1):
    print (i, '! ->', fatorial(i))
